### PR TITLE
test(apps_test): add more specs for "deis run"

### LINKED ISSUE #204: No complex command tests for "deis run"
The existing tests mostly run `env`, but there should be an example of a command that itself contains dashes and flags, so the `--` style syntax can be validated:
```console
$ deis run -a myapp -- ls -alh && echo "-- dashes --"
```

### PR DESCRIPTION
**Pull request by mboersma**
This covers some commands that caused problems in the past.

~~This also necessarily includes the change from #211. I will rebase it after that merges.~~ Rebased.

Requires deis/controller#751.

Closes #204.

### CODE REVIEW

**Comment by helgi** [comment]
Should we throw in a `run` calling a fake binary to test a failure scenario?

**Comment by mboersma** [comment]
@helgi I added a test for a non-existent command--is that what you had in mind?

But it uncovered that `deis apps:run /usr/bin/boguscmd` returns 0 and prints `No such file or directory` to stdout, not stderr. Those seem like bugs to me so I'll enter them at workflow-cli.

**Comment by helgi** [comment]
@mboersma yes sir!

As far as the bug, could also be related to the new way we do deis run in the API

**Comment by helgi** [comment]
Oh nevermind, I see why CLI

File: tests/apps_test.go [hunk 1]
```diff
@@ -190,11 +190,43 @@ var _ = Describe("deis apps", func() {
...
+				sess, err := cmd.Start("deis apps:run --app=%s dd if=/dev/urandom bs=3072 count=1000", &user, app.Name)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(sess, (settings.MaxEventuallyTimeout)).Should(Exit(0))
+				Expect(len(sess.Out.Contents())).To(BeNumerically(">=", 3072000))
+			})
+
+			Specify("that user can't run a bogus command in that app's environment", func() {
+				sess, err := cmd.Start("deis apps:run --app=%s /usr/bin/boguscmd", &user, app.Name)
+				Expect(err).NotTo(HaveOccurred())
+				// TODO: should this have gone to stderr instead?
```
Comment (vdice) [inline tests/apps_test.go]: have you answered your question here @mboersma ?

**Comment by vdice** [comment]
Added a comment re: the `TODO` note but not blocking for this PR; code LGTM.  (Do note that the jenkins job failure needs investigation...)

File: tests/apps_test.go (hunk 1 above)
Comment (mboersma) [inline tests/apps_test.go]: See deis/workflow-cli#69.

**Pull request merged**